lib/classes/Variables.js

```diff
@@ -216,7 +219,7 @@ class Variables {
     deepProperties.forEach(subProperty => {
       if (typeof valueToPopulate === 'undefined') {
         valueToPopulate = {};
-      } else {
+      } else if (subProperty !== '') {
```

Comment:
You bet.  The additional conditional will resolve to true in the case of ${self:}, any multiple dots in a row (e.g. ${[...]attr1..attrNext[...]}), or a trailing dot (e.g. ${[...]attr1.attrNext.}).  In each case, '""' in valueToPropagate will evaluate to false (I think? Actually, maybe I should add that check because I think obj[''] = {} is valid code, technically speaking) and the empty string indicates that the current value of valueToPropagate should continue to be the current value.  Another way to state this is that the empty string does not represent an attribute to traverse, therefore no traversal should occur and the current state continues to be the specified valid state.

Apologies for lack of formatting and whatnot, on the phone this morning and don't have a grave'.